### PR TITLE
270 implement gulp jscs

### PR DESCRIPTION
**Pull request by scott-dickson**
I have applied jscs to the current version of files using a custom version of crockford.

Feedback welcome

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -170,7 +171,7 @@ gulp.task('ractive-build-decorators', function () {
 	return gulp.src([
 		'./src/decorators/**/*.js',
 		'!./src/decorators/**/*.steps.js'
-	])
+		])
```
Comment (ivanwills) [inline gulpfile.js:174]: Either the two lines above should be indented or this line shouldn't and the 3 lines below unindented.

File: src/components/ux-button/ux-button.steps.js [hunk 2]
```diff
@@ -32,22 +32,21 @@ module.exports = function () {
...
+						callback.fail(e.name + ' ' + e.message);
+					}
 				}
+			);
 
-			}
-		);
-
-	});
+		});
```
Comment (ivanwills) [inline src/components/ux-button/ux-button.steps.js]: If you line the curly braces (}) you should also align the round brackets ie split this line into to.

File: src/components/ux-tabarea/ux-tabarea.steps.js [hunk 3]
```diff
@@ -26,5 +26,5 @@ module.exports = function () {
 	this.Given(/^I have loaded component "([^"]*)" with use case "([^"]*)"$/,
 		function (componentName, useCase, callback) {
 			this.client.loadComponentWithUseCase(componentName, useCase, callback);
-	});
+		});
```
Comment (ivanwills) [inline src/components/ux-tabarea/ux-tabarea.steps.js]: If you line the curly braces (}) you should also align the round brackets ie split this line into to.

File: src/components/ux-thumbnail/ux-thumbnail.steps.js [hunk 4]
```diff
@@ -16,6 +16,6 @@ module.exports = function () {
 	this.Given(/^I have loaded component "([^"]*)" with use case "([^"]*)"$/,
 		function (componentName, useCase, callback) {
 			this.client.loadComponentWithUseCase(componentName, useCase, callback);
-	});
+		});
```
Comment (ivanwills) [inline src/components/ux-thumbnail/ux-thumbnail.steps.js]: If you line the curly braces (}) you should also align the round brackets ie split this line into to.

File: src/decorators/tooltip/tooltip.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /*jshint unused:false */
 /*jshint -W025 */
 
-function(node, options) {
+function tooltip(node, options) {
```
Comment (ivanwills) [inline src/decorators/tooltip/tooltip.js]: This IS an anonymous function that will be assigned during build, I'm not sure giving it a name is valid (see how gulp task works). It is possible there is a better way to set this up but giving it a name is solving the wrong problem.

File: gulpfile.js (hunk 1 above)
Comment (scott-dickson) [inline gulpfile.js:174]: Good pick up there @ivanwills thanks!  Added indents to the lines above.

File: src/components/ux-tabarea/ux-tabarea.steps.js (hunk 3 above)
Comment (scott-dickson) [inline src/components/ux-tabarea/ux-tabarea.steps.js]: Thanks @ivanwills that looks better.

File: src/components/ux-button/ux-button.steps.js (hunk 2 above)
Comment (scott-dickson) [inline src/components/ux-button/ux-button.steps.js]: Thanks @ivanwills that looks better.

File: src/components/ux-thumbnail/ux-thumbnail.steps.js (hunk 4 above)
Comment (scott-dickson) [inline src/components/ux-thumbnail/ux-thumbnail.steps.js]: Thanks @ivanwills that looks better.

File: src/decorators/tooltip/tooltip.js (hunk 5 above)
Comment (scott-dickson) [inline src/decorators/tooltip/tooltip.js]: Dead right.  Clearly wasn't thinking.

**Pull request merged**